redlock/__init__.py

```diff
@@ -77,23 +88,39 @@ def lock(self, resource, ttl):
         # drift for small TTLs.
         drift = int(ttl * self.clock_drift_factor) + 2
 
+        redis_errors = list()
         while retry < self.retry_count:
             n = 0
             start_time = int(time.time() * 1000)
+            redis_errors.clear()
```

Comment:
http://stackoverflow.com/questions/10446839/does-dictionarys-clear-method-delete-all-the-item-related-objects-from-memory: `clear` is supported in Python 2 and `del redis_errors[:]` raises a `TypeError: unhashable type` as follows

```
Python 2.7.10 (default, Oct 23 2015, 18:05:06)
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> d = {"a":"b"}
>>> del d[:]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unhashable type
>>> d.clear()
>>> d
{}
```